digicampipe/io/generate_slow_containers.py

```diff
@@ -0,0 +1,203 @@
+property of the slow data.
+In case you want to use slow data files with a different format than the one 
+currently used, empty the digicampipe/tests/resources/slow/ folder, place the 
+slow data you want to use in it an run this script.
+"""
+
+
+def unit_parser(column):
+    unit = str(column.unit)
+    if ',' in unit:
+        unit = 'None'
+    m = re.match('(?:\[)?(?P<unit>[\w]+)(?:\])?', unit)
+    unit = m.group("unit")
+    if unit.lower() == 'lsb' or unit.lower() == 'boolean':
+        return 'bit'
+    if unit.lower() == 'sec':
+        return 'second'
+    if unit.lower() == 'days':
+        return 'day'
+    if unit.lower() == 'celsius':
+        return 'Celsius'
+    return unit
+
+
+def type_parser(column):
+    if len(column.dtype.shape):
+        return 'ndarray'
+    return column.dtype.name
+
+
+def description_parser(comment):
+    ret = {}
+    for line in comment.replace('\n', '').split('//'):
+        if len(line) > 0:
+            tmp = line.split(' - ')
+            ret[tmp[0]] = tmp[1]
+    return ret
+
+
+def columns_info(hdu):
+    res = {}
+    descriptions = None
+    if 'COMMENT' in hdu.header.keys():
+        comment = hdu.header['COMMENT'].__str__()
+        descriptions = description_parser(comment)
+    for column in hdu.columns:
+        description = column.name
+        if descriptions is not None and column.name in descriptions:
+            description = descriptions[column.name]
+        res[column.name] = {
+            'type': type_parser(column),
+            'units': unit_parser(column),
+            'desc': description
+        }
+    return res
+
+
+def get_class_name(path):
+    filename = os.path.basename(path)
+    m = re.match('(?:slow_)?(?P<class>[\w]+?)_[\d\_]+\.fits', filename)
+    class_name = m.group("class")
+    return class_name
+
+
+def generate_container_class(path_file, file=sys.stdout):
+    class_name = get_class_name(path_file)
+    print('class %sContainer(Container):' % class_name, file=file)
+    with fits.open(path_file) as hdul:
+        results = columns_info(hdul[1])
+        lower_dict = dict((k.lower(), v) for k, v in results.items())
+        if 'timestamp' not in lower_dict.keys():
+            print('Warning in generate_container_class: no timestamp',
+                  'column found for class', class_name,
+                  'in file', path_file, file=file)
+        for key in sorted(lower_dict):
+            values = lower_dict[key]
+            if str(values['units']) not in ['none', 'None', None]:
+                data_printed = (key, values['type'],
+                                values['desc'], values['units'])
+                print('    %s = Field(%s, "%s", unit=u.%s)'
+                      % data_printed, file=file)
+            else:
+                print('    %s = Field(%s, "%s")'
+                      % (key, values['type'], values['desc']), file=file)
+    print('\n', file=file)
+    return class_name
```

Comment:
Sorry I don't understand why this function does : 
1. read class_name
2. do bunch of stuff with hdu
3. return class_name

Should it not return the container ?

Reply:
ah I see... the function should write some data into a file write ? If so I suggest renaming `generate_` to `write_` or something else. generate indicates (to my opinion) that it creates the object (inside memory)

Reply:
Also file=sys.stdout as default value is maybe miss leading no? because in which situation would you use that except  for debugging?

Reply:
the goal of the generate_... functions is to create parts of the slow_container.py file.
It is indeed defaulted to sys.stdout for debug (and to avoid to recreate the slow_container.py file accidentally).
I can change the default value if you want.

Reply:
ok. then I think having no default value is better...
because the user would just run the function and wonder what it does except just printing stuff. 
Forcing the user to choose a file also diminishes the risk of overwriting, no?
As you prefer.